File: Math/4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.java
class Solution {
    public long sumAndMultiply(int n) {
        int num =0; 
        int sum =0,base=1;
        while(n>0){
            int r = n%10;
            if(r!=0) {
                num += base*r;
                base *=10;
            }
            sum += r;
            n /=10;
        }
        return (long)sum * num;
    }
}

File: Math/4135-concatenate-non-zero-digits-and-multiply-by-sum-i/concatenate-non-zero-digits-and-multiply-by-sum-i.py
class Solution:
    def sumAndMultiply(self, n: int) -> int:
        num =0
        sum =0
        base =1
        while n:
            r = n%10
            if r :
                num += r*base
                base *=10
            sum +=r
            n = n//10
        return sum * num